homework 1/ReaderWriterLock/SharedResourceRwLock.cs

```diff
@@ -1,19 +1,32 @@
+using System.Threading;
+
 namespace ReaderWriterLock;
 
 public class SharedResourceRwLock : SharedResourceBase
 {
+    private string value;
+    private readonly ReaderWriterLockSlim locker = new();
+    
     public override void Write(string data)
     {
-        throw new System.NotImplementedException();
+        locker.EnterWriteLock();
+        value = data;
+        locker.ExitWriteLock();
```

Comment:
```suggestion
        locker.EnterWriteLock();
        try
        {
            value = data;
        }
        finally
        {
            locker.ExitWriteLock();
        }
```
чтобы даже при ошибке гарантировать выход из лока